### PR TITLE
[14.0] [mig] web_translate_dialog

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

**Comment by dsolanki-initos** [comment]
@ferranmateu Will you please review this PR?

**Review by claudiagn — COMMENTED** [review]
It looks like these:
captura_6

And it should look like these:
captura_5

The thing is that you can translate the name, but not all the fields, and also the field is duplicated.

**Comment by dsolanki-initos** [comment]
Hi @claudiagn ,
can you please re-check it?

**Review by simahawk — CHANGES_REQUESTED** [review]
same remarks of https://github.com/OCA/web/pull/1801

**Comment by dsolanki-initos** [comment]
I have fixed those remarks it would be great if it is reviewd and merge it ASAP.

**Comment by dsolanki-initos** [comment]
I have fixed those remarks it would be great if it is reviewd and merge it ASAP.

**Comment by dsolanki-initos** [comment]
Pre-commit is failed as I need to overide the method to fix the above issue.

**Comment by dsolanki-initos** [comment]
It would be great if we get the review here

**Comment by dsolanki-initos** [comment]
It would be great if we get the review here.

**Comment by dsolanki-initos** [comment]
It would be great if we get the reviews here.

**Comment by dsolanki-initos** [comment]
It would be great if we get the reviews here.

**Comment by dsolanki-initos** [comment]
It would be great if we get the reviews here.

**Comment by dsolanki-initos** [comment]
It would be great if we get the reviews here.

File: web_translate_dialog/models/base.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models
+
+
+class IrTranslation(models.Model):
+    _inherit = "ir.translation"
+
+    @api.model
```
Comment (gfcapalbo) [inline web_translate_dialog/models/base.py]: it's appropriately model. it makes sense for this function. I agree with this.

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 2]
```diff
@@ -0,0 +1,378 @@
+/* Copyright 2012 Guewen Baconnier (Camptocamp SA)
+   Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+   Copyright 2021 Dhara Solanki <dhara.solanki@initos.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define("web_translate_dialog.translate_dialog", function (require) {
+    "use strict";
+
+    var core = require("web.core");
```
Comment (gfcapalbo) [inline web_translate_dialog/static/src/js/web_translate_dialog.js:8]: all requires strictly needed no cruft here :+1:

**Review by gfcapalbo — APPROVED** [review]
:+1 LGTM, tested and read, could not find problems.

**Comment by dsolanki-initos** [comment]
It would be nice if we get the reviews here

**Comment by dsolanki-initos** [comment]
Hi @claudiagn 
Can you please re-check the PR

**Comment by dsolanki-initos** [comment]
Hi @claudiagn @simahawk 
Can you please re-check the PR

**Comment by simahawk** [comment]
I got this on runbot http://3500856-1812-322b67.runbot1-3.odoo-community.org/web?debug=assets#id=23&action=278&model=product.template&view_type=form&cids=1&menu_id=174
```
Error:
TypeError: initialValue is undefined

http://3500856-1812-322b67.runbot1-3.odoo-community.org/web_translate_dialog/static/src/js/web_translate_dialog.js:204
Traceback:
_onSave/<@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web_translate_dialog/static/src/js/web_translate_dialog.js:204:21
_onSave@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web_translate_dialog/static/src/js/web_translate_dialog.js:202:67
_setButtonsTo/</<@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web/static/src/js/core/dialog.js:322:44
dispatch@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web/static/lib/jquery/jquery.js:5183:27
add/elemData.handle@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web/static/lib/jquery/jquery.js:4991:28
```
@dsolanki-initos could you check?

**Comment by dsolanki-initos** [comment]
> I got this on runbot http://3500856-1812-322b67.runbot1-3.odoo-community.org/web?debug=assets#id=23&action=278&model=product.template&view_type=form&cids=1&menu_id=174
> 
> ```
> Error:
> TypeError: initialValue is undefined
> 
> http://3500856-1812-322b67.runbot1-3.odoo-community.org/web_translate_dialog/static/src/js/web_translate_dialog.js:204
> Traceback:
> _onSave/<@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web_translate_dialog/static/src/js/web_translate_dialog.js:204:21
> _onSave@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web_translate_dialog/static/src/js/web_translate_dialog.js:202:67
> _setButtonsTo/</<@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web/static/src/js/core/dialog.js:322:44
> dispatch@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web/static/lib/jquery/jquery.js:5183:27
> add/elemData.handle@http://3500856-1812-322b67.runbot1-3.odoo-community.org/web/static/lib/jquery/jquery.js:4991:28
> ```
> 
> @dsolanki-initos could you check?

Hi @simahawk 
I am not able to login with admin credentail with this link you shared. 
I've double verified it locally I am not getting such error. Can you please share video ?

**Comment by dsolanki-initos** [comment]
Hi @gfcapalbo  @simahawk
Can you please re-check the PR?

**Comment by dsolanki-initos** [comment]
Hi @gfcapalbo @simahawk
Can you please re-check the PR?

**Comment by dsolanki-initos** [comment]
Hi @gfcapalbo @simahawk
Can you please re-check the PR?

**Comment by dsolanki-initos** [comment]
Hi @gfcapalbo @simahawk
Can you please re-check the PR?

**Comment by dsolanki-initos** [comment]
Hi @gfcapalbo @simahawk
Can you please re-check the PR?

**Comment by dsolanki-initos** [comment]
Hi @simahawk
I've verified it locally I am not getting an error. Can you please share a video?

File: web_translate_dialog/tests/test_web_translate_dialog.py [hunk 3]
```diff
@@ -6,36 +6,64 @@
 class TestWebTranslateDialog(TransactionCase):
     def setUp(self):
         super(TestWebTranslateDialog, self).setUp()
-        self.product = self.env["product.template"].create({"name": "English Name"})
+        self.env["res.lang"]._activate_lang("de_DE")
+        self.partner_title = self.env["res.partner.title"].create({"name": "Doctor"})
```
Comment (simahawk) [inline web_translate_dialog/tests/test_web_translate_dialog.py:9]: Can you move this to `SavepointCase.setUpClass`? This code runs for all tests.

**Review by simahawk — CHANGES_REQUESTED** [review]
question: don't we need to fwd port this? https://github.com/OCA/web/pull/1974

**Comment by simahawk** [comment]
> Hi @simahawk I've verified it locally I am not getting an error. Can you please share a video?

sorry, I have no time now. I trust you've done proper testing :wink:

**Comment by dsolanki-initos** [comment]
@simahawk could please give approval to this pull?

**Comment by simahawk** [comment]
> @simahawk could please give approval to this pull?

have you seen my comment above?

**Comment by dsolanki-initos** [comment]
@albariera could you please check this pull?

**Comment by dsolanki-initos** [comment]
@simahawk could you please give approval to this pull?

**Comment by simahawk** [comment]
@dsolanki-initos you pinged me already several times but my comments are still not attended, not even answered :man_shrugging: 


image

**Comment by bosd** [comment]
Interesting PR!

**Pull request closed without merging**